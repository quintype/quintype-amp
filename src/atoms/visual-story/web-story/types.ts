import { Story } from "../../../types/story";
import { Config } from "../../../types/config";

export interface WebStoryTypes {
  children: JSX.Element[] | JSX.Element | React.ReactChildren | React.ReactChild;
  story: Story;
  config: Config;
}
