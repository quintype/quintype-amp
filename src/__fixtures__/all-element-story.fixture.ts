const allElementsStory = {
  "updated-at": 1585646852797,
  seo: {
    "claim-reviews": {
      story: null
    }
  },
  "assignee-id": 190071,
  "author-name": "tester",
  tags: [],
  headline: "Doctor Speaks Out to Save Elderly Patients from the Coronavirus",
  "storyline-id": null,
  votes: {},
  "story-content-id": "05066b94-784b-4913-971b-c01a1bcfb43d",
  slug: "news/health/doctor-speaks-out-to-save-elderly-patients-from-the-coronavirus",
  "linked-stories": {
    "78bae455-84d9-4217-9208-6927ed5989e1": {
      "author-name": "Shitesh Jain",
      headline: "new default story headline size is being increased upto c",
      "story-content-id": "78bae455-84d9-4217-9208-6927ed5989e1",
      slug: "current-affairs/new-default-story",
      sections: [
        {
          id: 6449,
          name: "Current Affairs",
          slug: "current-affairs",
          data: null
        }
      ],
      "hero-image-metadata": {
        width: 760,
        height: 506,
        "mime-type": "image/jpeg"
      },
      "hero-image-s3-key": "ace/2019-06/051658dc-43f0-4945-92bd-f2da2a206e40/panda.jpg",
      url: "https://ace-web.qtstage.io/current-affairs/new-default-story",
      "content-updated-at": 1559804236157,
      "author-id": 234915,
      "first-published-at": 1554788924047,
      authors: [
        {
          id: 234915,
          name: "Shitesh Jain",
          slug: "shitesh-jain-4",
          "avatar-url": "https://lh3.googleusercontent.com/a-/AAuE7mBO5TOGmzwz6_e4-1i0wQ2-Z8qEnkyp3GWL-aGm",
          "avatar-s3-key": null,
          "twitter-handle": null,
          bio: null,
          "contributor-role": {
            id: 873,
            name: "Author"
          }
        }
      ]
    }
  },
  "last-published-at": 1585646855541,
  subheadline: "subtitle",
  alternative: {
    home: {
      default: {
        headline: "Alternative headline for the story",
        "hero-image": {
          "member-id": 934395,
          "hero-image-extracted-data": null,
          "uploaded-at": 1584677708858,
          "hero-image-metadata": {
            "original-url":
              "https://image.cnbcfm.com/api/v1/image/106329196-1578602903227gettyimages-1194751175.jpeg?v=1578603661",
            width: 1611,
            height: 1169
          },
          "hero-image-attribution": "Attribution",
          "hero-image-alt-text": "Alt text",
          "hero-image-s3-key":
            "ace/2020-03/4f3bfc9a-209c-4f5d-a59e-616ac4d1f7cc/106329196_1578602903227gettyimages_1194751175.jpeg",
          "creator-id": null,
          "hero-image-caption": "Trump",
          "extracted-data": null,
          "hero-image-url":
            "https://thumbor-stg.assettype.com/ace/2020-03/4f3bfc9a-209c-4f5d-a59e-616ac4d1f7cc/106329196_1578602903227gettyimages_1194751175.jpeg",
          "temporary-hero-image-key": null
        }
      }
    }
  },
  sections: [
    {
      "domain-slug": null,
      slug: "health",
      name: "Health",
      "section-url": "https://ace-web.qtstage.io/news/health",
      id: 11181,
      "parent-id": 38586,
      "display-name": "Health",
      collection: {
        slug: "health",
        name: "Health",
        id: 15603
      },
      data: null
    }
  ],
  "story-audio": {
    "s3-key": "ace/story-audio/2020-03/05066b94-784b-4913-971b-c01a1bcfb43d/.820db8e1-73e8-4e9b-a5ad-063b0f09acb0.mp3"
  },
  "read-time": 6,
  "access-level-value": null,
  "content-created-at": 1584680032615,
  "owner-name": "tester",
  "custom-slug": null,
  "push-notification": null,
  "publisher-id": 97,
  "hero-image-metadata": {
    width: 3456,
    "focus-point": [1513, 1041],
    height: 1999
  },
  comments: null,
  "word-count": 1347,
  entities: {},
  "published-at": 1585646855541,
  "is-live-blog": false,
  "breaking-news-linked-story-id": null,
  "storyline-title": null,
  summary: null,
  "push-notification-title": null,
  "external-id": null,
  "canonical-url": null,
  "is-amp-supported": true,
  autotags: [],
  "linked-entities": [],
  status: "published",
  "hero-image-attribution": null,
  "bullet-type": "123",
  id: "05066b94-784b-4913-971b-c01a1bcfb43d",
  "hero-image-alt-text": "Alt text",
  "hero-image-s3-key": "ace/2018-02/b2e9d05a-d633-4046-9433-3cce782b2638/hero_hd.jpg",
  contributors: [],
  "associated-series-collection-ids": [],
  cards: [
    {
      "story-elements": [
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/77b3c9df-2821-4587-a952-d6c14879b1df",
          type: "text",
          "family-id": "cda34053-40da-4aa5-854c-fe43d3e634ed",
          title: "",
          id: "77b3c9df-2821-4587-a952-d6c14879b1df",
          metadata: {},
          subtype: "summary",
          text:
            "<p>In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest. [If you have been in close contact with a person who has tested positive and you are asymptomatic, experts recommend staying home for fourteen days after that contact.] If I don’t get tested, I’ll stay home for seven days. But if I do get tested and I test negative, I could go back.</p>"
        },
        {
          description: "",
          "page-url": "/story/af018d6b-2166-4344-b167-40405a148e41/element/a50b2136-c18b-4dd9-a2b2-0df96e2c253a",
          type: "composite",
          "family-id": "88e7df74-fe20-42ad-b5aa-54de86108fc4",
          "story-elements": [
            {
              description: "",
              "image-metadata": {
                width: 656,
                height: 280
              },
              type: "image",
              "family-id": "fae56324-4c23-45b7-baac-81da10ae5ff6",
              "image-attribution": "Attribution of an imitation game",
              title: "Chase: An Imitation Game - A Story On Dharavi's Fake Leather Bags",
              id: "5c375339-5fb0-481a-a811-ae271d894b3b",
              "alt-text": "Alt text",
              "image-s3-key":
                "ace/2020-03/ad503d63-4d05-41e6-9722-fdfe3acb2405/handwashing_8_wide_eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg",
              metadata: {},
              subtype: null
            },
            {
              description: "",
              "image-metadata": {
                width: 326,
                height: 155
              },
              type: "image",
              "family-id": "953a1d2c-d52d-4b13-8af5-969e74813197",
              "image-attribution": "Ramsharan Gorur Jayaraman",
              title: "Caption",
              id: "48b14295-99ed-4fc8-ad99-4cb1d85a29ae",
              "alt-text": "Alt text",
              "image-s3-key": "ace/2019-06/051658dc-43f0-4945-92bd-f2da2a206e40/panda.jpg",
              metadata: {},
              subtype: null
            }
          ],
          title: "",
          id: "a50b2136-c18b-4dd9-a2b2-0df96e2c253a",
          metadata: {
            type: "slideshow"
          },
          subtype: "image-gallery"
        },
        {
          description: "",
          "page-url": "/story/af018d6b-2166-4344-b167-40405a148e41/element/db5dd415-e4e1-41f8-83af-48f771fe22ec",
          type: "composite",
          "family-id": "015f0be5-428f-47f2-86ff-fba42be6c77b",
          "story-elements": [
            {
              description: "",
              "image-metadata": {
                width: 1200,
                height: 520
              },
              type: "image",
              "family-id": "6532027a-f855-474c-8651-dbdd96de3b39",
              "image-attribution": "",
              title: "Are college interviews now a saffron litmus test?",
              id: "36452445-2155-496a-b471-de48870b713a",
              "alt-text": "Alt text",
              "image-s3-key": "ace/2019-06/051658dc-43f0-4945-92bd-f2da2a206e40/panda.jpg",
              metadata: {},
              subtype: null
            },
            {
              description: "",
              "image-metadata": {
                width: 3128,
                height: 3910,
                "focus-point": [350, 3166]
              },
              type: "image",
              "family-id": "e343544a-93b4-43d8-bc52-2c3a63adeac7",
              "image-attribution": "unsplash",
              title: "Traffic cones",
              id: "0898278c-9fac-4685-a0ae-d9caf8c1e7a7",
              "alt-text": "Alt text",
              "image-s3-key":
                "ace/2020-03/ad503d63-4d05-41e6-9722-fdfe3acb2405/handwashing_8_wide_eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg",
              metadata: {},
              subtype: null
            }
          ],
          title: "",
          id: "db5dd415-e4e1-41f8-83af-48f771fe22ec",
          metadata: {
            type: "gallery"
          },
          subtype: "image-gallery"
        },
        {
          description: "",
          "embed-js":
            "PGlmcmFtZSBoZWlnaHQ9IjI2NSIgc3R5bGU9IndpZHRoOiAxMDAlOyIgc2Nyb2xsaW5nPSJubyIgdGl0bGU9ImxpZ2h0IGJ1bGIiIHNyYz0iaHR0cHM6Ly9jb2RlcGVuLmlvL3ZsYWRpc3NsYXd3dy9lbWJlZC9wb2p2enFaP2hlaWdodD0yNjUmdGhlbWUtaWQ9ZGFyayZkZWZhdWx0LXRhYj1odG1sLHJlc3VsdCIgZnJhbWVib3JkZXI9Im5vIiBhbGxvd3RyYW5zcGFyZW5jeT0idHJ1ZSIgYWxsb3dmdWxsc2NyZWVuPSJ0cnVlIiBsb2FkaW5nPSJsYXp5Ij4KICBTZWUgdGhlIFBlbiA8YSBocmVmPSdodHRwczovL2NvZGVwZW4uaW8vdmxhZGlzc2xhd3d3L3Blbi9wb2p2enFaJz5saWdodCBidWxiPC9hPiBieSB2bGFkaXNzbGF3dwogICg8YSBocmVmPSdodHRwczovL2NvZGVwZW4uaW8vdmxhZGlzc2xhd3d3Jz5AdmxhZGlzc2xhd3d3PC9hPikgb24gPGEgaHJlZj0naHR0cHM6Ly9jb2RlcGVuLmlvJz5Db2RlUGVuPC9hPi4KPC9pZnJhbWU+",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/5bf7ba6d-9a51-41f2-b919-47539c4a5acd",
          type: "jsembed",
          "family-id": "f564d440-c969-4c35-9c5e-ddb032e22b05",
          title: "",
          id: "5bf7ba6d-9a51-41f2-b919-47539c4a5acd",
          metadata: {},
          subtype: null
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/28ae64b8-cb6d-4410-bc4e-6021fac5427f",
          type: "text",
          "family-id": "07ba0b67-1359-40ae-b566-bbca9188692a",
          title: "",
          id: "28ae64b8-cb6d-4410-bc4e-6021fac5427f",
          metadata: {},
          subtype: null,
          text:
            '<p>Rachael Bedard is a geriatrician who works on <ins><a href="https://www.newyorker.com/tag/rikers-island">Rikers Island</a></ins>, caring for the oldest and sickest people in the New York City jail system. This week, as <ins><a href="https://www.newyorker.com/tag/bill-de-blasio">Mayor Bill de Blasio</a></ins> banned large gatherings and considered a citywide lockdown, Bedard and a few of her colleagues spoke out on behalf of their patients. “We need to take the unprecedented step TODAY of providing urgent release to everyone in the jails who is at risk of serious morbidity and mortality from COVID,” Bedard wrote on Twitter, on Wednesday. That day, local news outlets reported that the <ins><a href="https://www.newyorker.com/tag/coronavirus">coronavirus</a></ins> had arrived on Rikers: an incarcerated person and a correction officer who worked at a security gate near the island’s entrance had both tested positive for <em>covid</em>-19. There are about fifty-four hundred people in the city’s jails, and more than nine hundred of them are fifty and older, according to the Board of Correction. Most have not been convicted of the charges against them; they are waiting for their cases to move through the courts.</p>'
        },
        {
          description: "",
          "image-metadata": {
            "original-url":
              "https://media.npr.org/assets/img/2020/03/20/handwashing-8_wide-eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg?s=1400",
            width: 1400,
            height: 787
          },
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/2c31b5ad-0189-4bad-841e-e105d738ca80",
          type: "image",
          "family-id": "c608c12f-f45b-4266-8a60-9de857ee92fd",
          "image-attribution": "Someone ",
          title: "Wash your hands",
          id: "2c31b5ad-0189-4bad-841e-e105d738ca80",
          "alt-text": "Alt text",
          "image-s3-key":
            "ace/2020-03/ad503d63-4d05-41e6-9722-fdfe3acb2405/handwashing_8_wide_eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg",
          metadata: {},
          subtype: null
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/4e1d3a72-0eec-40c1-8b89-8a84e09d5d81",
          type: "text",
          "family-id": "9a2a4731-55aa-478d-bc6c-805c24f2b654",
          title: "",
          id: "4e1d3a72-0eec-40c1-8b89-8a84e09d5d81",
          metadata: {},
          subtype: null,
          text:
            "<p>By Thursday, Bedard, who is thirty-seven, had a low-grade fever and stayed home from work; she has not yet been tested for <em>covid</em>-19. When we spoke on the phone that afternoon, she sometimes had to pause because, she said, “I’m a little short of breath.” That day, de Blasio announced plans to release forty people from the jail system in light of the coronavirus outbreak, a number that Bedard and advocates say is not sufficient. When we spoke, I asked Bedard about her job, the pressures she faces now, and what she and her colleagues are doing to try to stop the spread of the virus inside the city’s jails. Our conversation has been edited for length and clarity.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/92dc4ff1-1a75-4e08-b029-68c5ef53accf",
          type: "text",
          "family-id": "4a00c340-f391-4e5c-ac0a-d7197c6fd0ef",
          title: "",
          id: "92dc4ff1-1a75-4e08-b029-68c5ef53accf",
          metadata: {
            content:
              "People talk about jails and prisons being incubators for infection. How does that work exactly on Rikers?",
            attribution: "Bedbard"
          },
          subtype: "blockquote",
          text:
            '<div><blockquote>People talk about jails and prisons being incubators for infection. How does that work exactly on Rikers?</blockquote><span class="attribution">Bedbard</span></div>'
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/57582578-1a3f-417a-9ab9-469d0dde12dd",
          type: "text",
          "family-id": "8ac48b48-fd26-4997-819d-8e91b8c35841",
          title: "",
          id: "57582578-1a3f-417a-9ab9-469d0dde12dd",
          metadata: {},
          subtype: null,
          text:
            "<p>Rikers Island is a mini city unto itself. It is not just detainees and guards. It is thousands of people coming on and off every day. And so, in the same way that the city is struggling with the complexity of its response, we are doing that on our smaller but almost more urgent scale because, like a nursing home, a jail is a perfect setup for an outbreak.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/9f842c63-b3c0-4426-b066-b37268286b58",
          type: "text",
          "family-id": "d3915837-3583-4ebb-908e-f26b5d62c7c4",
          title: "",
          id: "9f842c63-b3c0-4426-b066-b37268286b58",
          metadata: {
            content:
              "Before covid19 appeared on Rikers Island, what was your job? What types of individuals were you caring for on a daily basis?",
            attribution: "The Newyorker"
          },
          subtype: "quote",
          text:
            '<div><blockquote>Before covid19 appeared on Rikers Island, what was your job? What types of individuals were you caring for on a daily basis?</blockquote><span class="attribution">The Newyorker</span></div>'
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/c9473912-184a-4f97-8093-90f677dbb250",
          type: "text",
          "family-id": "c3400aa4-b640-488d-898c-5d1f8e460877",
          title: "",
          id: "c9473912-184a-4f97-8093-90f677dbb250",
          metadata: {},
          subtype: null,
          text:
            "<p>I’m the senior director of the geriatrics and complex-care service. My specialty is taking care of people who are older and sick and then also coördinating with the courts to come up with creative solutions to get these folks out of custody as often as possible. I run a team of two other clinicians—a nurse and a nurse practitioner—social workers, and reëntry specialists. And we, as a team, follow the hundred-and-eighty-five oldest and sickest patients in the jail system through the facilities, as they go to the hospital, and then for a period of time after they leave.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/3d686835-0a6e-4ac9-87f5-51b5c161a620",
          type: "text",
          "family-id": "a69fe60a-e3c2-48a9-ac38-e6dccd4d3bec",
          title: "",
          id: "3d686835-0a6e-4ac9-87f5-51b5c161a620",
          metadata: {
            question: "<p><strong>Tell me about some of your patients. What health problems do they have?</strong></p>",
            answer:
              "<p>First of all, I think there’s this incredible misconception about who’s sitting in jail. I have taken care of, in my three-and-a-half-year jail career, multiple patients over ninety. I currently have a patient over ninety.</p>"
          },
          subtype: "q-and-a",
          text:
            '<div><div class="question"><p><strong>Tell me about some of your patients. What health problems do they have?</strong></p></div><div class="answer"><p>First of all, I think there’s this incredible misconception about who’s sitting in jail. I have taken care of, in my three-and-a-half-year jail career, multiple patients over ninety. I currently have a patient over ninety.</p></div></div>'
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/8a17dfa6-e7b7-431b-83d8-007edee90c1a",
          type: "text",
          "family-id": "2266eee0-aeef-46bd-a2d2-e4790bdddb70",
          title: "",
          id: "8a17dfa6-e7b7-431b-83d8-007edee90c1a",
          metadata: {},
          subtype: null,
          text:
            "<p>Jails are a place where you see an incredible concentration of pathology for a variety of reasons. On the whole, our population has poor access to primary preventative health care. These are folks who have been living on the margins. Poor socioeconomic status, homelessness, etc., are obviously incredibly prevalent for our guys.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/2a3f1e74-df4b-4c8e-99de-57a69366b77f",
          type: "youtube-video",
          "family-id": "8256f5d1-5703-4c3f-8fc0-a6c4f44e531c",
          title: "",
          id: "2a3f1e74-df4b-4c8e-99de-57a69366b77f",
          url: "https://www.youtube.com/watch?v=jlEbFvGwcUc",
          "embed-url": "https://www.youtube.com/embed/jlEbFvGwcUc",
          metadata: {},
          subtype: null
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/40f27611-f0d0-4e4d-853a-f36c595ee21f",
          type: "text",
          "family-id": "0668dc6d-ffea-45a6-8531-5403fdf0a8e8",
          title: "",
          id: "40f27611-f0d0-4e4d-853a-f36c595ee21f",
          metadata: {},
          subtype: "question",
          text: "<p><strong>You have a patient on Rikers who is older than ninety?</strong></p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/19a1acd5-3777-491d-af15-9032ce5cd69b",
          type: "text",
          "family-id": "4117ebe1-ebf4-4d8b-bf72-2acd23c98a92",
          title: "",
          id: "19a1acd5-3777-491d-af15-9032ce5cd69b",
          metadata: {},
          subtype: "answer",
          text:
            "<p>Yes. This is a guy currently being held in pretrial detention. I have multiple patients in their eighties. I have patients who are A.D.L.-impaired—which means that they cannot attend to their own activities of daily living and would be nursing-home- or assisted-living-eligible if they were in the community—who need daily assistance with really basic tasks like toileting and changing themselves. I have patients who are paraplegic with significant wounds that require daily wound care. I have multiple patients who have active cancer and are getting chemotherapy and are both sick and immunosuppressed from their medication.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/8049ba24-3194-406b-8d01-ba2919ffd522",
          type: "text",
          "family-id": "ac718a86-32a9-4bd1-8fa3-524dc7004f4c",
          title: "",
          id: "8049ba24-3194-406b-8d01-ba2919ffd522",
          metadata: {},
          subtype: null,
          text:
            "<p>It has been my prime concern over the last three years to prevent in-detention mortality whenever we can see it coming. And we do that through compassionate-release work, and we’ve been relatively successful. I’ve been involved in requests to expedite release for over a hundred patients. We know that at least sixty or seventy of those have gone out to the community. At least fifteen or twenty people died within weeks of release, suggesting that they would have otherwise died while in custody.</p><p>The mortality statistics at Rikers have gone down dramatically in the last couple of years. We only had three deaths in the system last year. And that’s not just the result of the fact that the population is decreased. It really reflects urgent and difficult work trying to get sick people released. We’re not going to be able to do that on a case-by-case basis if the virus spreads the way we anticipate it spreading.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/0ee36b0b-fa30-4e84-9d1e-d3b26edb2524",
          type: "text",
          "family-id": "ff2ae3dc-b738-428a-9dff-880c1879b686",
          title: "",
          id: "0ee36b0b-fa30-4e84-9d1e-d3b26edb2524",
          metadata: {
            content: "124567",
            attribution: "Cases of Covid-19 worldwide"
          },
          subtype: "bigfact",
          text:
            '<div><div class="bigfact-title">124567</div><div class="bigfact-description">Cases of Covid-19 worldwide</div></div>'
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/3e1c4198-51e0-4528-9360-6c48ac865f6f",
          type: "text",
          "family-id": "292436f8-b296-45e9-bc95-005398654779",
          title: "",
          id: "3e1c4198-51e0-4528-9360-6c48ac865f6f",
          metadata: {},
          subtype: null,
          text:
            "<p>Right now, one of the major priorities is not introducing <em>covid</em> into the jails more than it’s already arrived. So the policy right now is, if you have any of the core symptoms—significant shortness of breath, cough, fever—you are supposed to not come to work. So there are definitely lots of people out, not just on the health-care-staff side but on the officer side.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/19d96101-04b2-400a-ab95-e7362280d8a1",
          type: "text",
          "family-id": "5f761f27-4721-459f-8ecd-733021d8d2ea",
          title: "",
          id: "19d96101-04b2-400a-ab95-e7362280d8a1",
          metadata: {
            content:
              "In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest."
          },
          subtype: "blurb",
          text:
            "<blockquote>In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest.</blockquote>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/5d3a41e0-3087-42ed-9b14-e5aab93b9751",
          type: "text",
          "family-id": "7dd2ba6f-a659-4091-a4ce-5cb3a4081d66",
          title: "",
          id: "5d3a41e0-3087-42ed-9b14-e5aab93b9751",
          metadata: {},
          subtype: null,
          text:
            "<p>Whenever you transport a guy in custody to the hospital, he goes with not just E.M.S. but with officers at his side who are theoretically supposed to be cuffing him to transport him. It puts those officers at risk. We contribute to an overburdened health-care crisis at our affiliated hospital. And we see patients die in custody.</p><p>It has been my prime concern over the last three years to prevent in-detention mortality whenever we can see it coming. And we do that through compassionate-release work, and we’ve been relatively successful. I’ve been involved in requests to expedite release for over a hundred patients. We know that at least sixty or seventy of those have gone out to the community. At least fifteen or twenty people died within weeks of release, suggesting that they would have otherwise died while in custody.</p><p>The mortality statistics at Rikers have gone down dramatically in the last couple of years. We only had three deaths in the system last year. And that’s not just the result of the fact that the population is decreased. It really reflects urgent and difficult work trying to get sick people released. We’re not going to be able to do that on a case-by-case basis if the virus spreads the way we anticipate it spreading.</p>"
        },
        {
          description: "",
          "embed-js":
            "PGlmcmFtZSBmcmFtZWJvcmRlcj0iMCIgd2lkdGg9IjY0MCIgaGVpZ2h0PSIzNjAiIHNyYz0iaHR0cHM6Ly93d3cuZGFpbHltb3Rpb24uY29tL2VtYmVkL3ZpZGVvL3g3dDluMTMiIGFsbG93ZnVsbHNjcmVlbiBhbGxvdz0iYXV0b3BsYXkiPjwvaWZyYW1lPg==",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/89d43010-36f3-427b-a344-ef1630dc9745",
          type: "jsembed",
          "family-id": "0c2c5683-37c5-4cbf-b51f-0fe43f408a23",
          title: "",
          id: "89d43010-36f3-427b-a344-ef1630dc9745",
          metadata: {
            "dailymotion-url": "https://www.dailymotion.com/video/x7t9n13",
            provider: "dailymotion-video",
            "video-id": "x7t9n13"
          },
          subtype: "dailymotion-video"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/31e2c51b-cd56-405f-8882-0c348a59fb7d",
          type: "text",
          "family-id": "3d6fba96-a6d9-430a-b781-f6bd5a7d7e50",
          title: "",
          id: "31e2c51b-cd56-405f-8882-0c348a59fb7d",
          metadata: {},
          subtype: null,
          text:
            "<p>Depopulating in this case is not letting a dozen guys out or two dozen guys out. It means pushing as hard as possible for hundreds of people to get out so that the jail population is decreased enough that you don’t just benefit the people who are released but you also benefit those who are left behind—and the staff who take care of them and the officers who take care of them.</p>"
        },
        {
          description: "",
          "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/946c7dfb-a6c8-4224-b2f6-8927afc183ef",
          type: "text",
          "family-id": "c5e76d29-517b-4619-924c-932cc82bf890",
          title: "",
          id: "946c7dfb-a6c8-4224-b2f6-8927afc183ef",
          metadata: {
            "linked-story-id": "78bae455-84d9-4217-9208-6927ed5989e1",
            "linked-story": {
              headline: "new default story headline size is being increased upto c",
              "story-content-id": "78bae455-84d9-4217-9208-6927ed5989e1",
              "highlighted-text": "",
              id: "78bae455-84d9-4217-9208-6927ed5989e1",
              "highlighted-headline": null
            }
          },
          subtype: "also-read",
          text: "new default story headline size is being increased upto c"
        },
        {
          id: "00000000-0000-0000-0000-000000000000",
          description: "",
          title: "",
          subtype: null,
          type: "text",
          text: '<p>text <a href="https://ace-web.qtstage.io/">promotional</a> message</p>',
          metadata: {
            "promotional-message": true
          }
        }
      ],
      "card-updated-at": 1585646849246,
      "content-version-id": "e480edd1-5e67-4081-8258-d64f7bbaa245",
      "card-added-at": 1584680085667,
      status: "draft",
      id: "98bb5875-957a-4435-90ef-4bb3caf79cc6",
      "content-id": "98bb5875-957a-4435-90ef-4bb3caf79cc6",
      version: 26,
      metadata: {
        "social-share": {
          shareable: false,
          title: "Doctor Speaks Out to Save Elderly Patients from the Coronavirus",
          message: null,
          image: {
            key:
              "ace/2020-03/ad503d63-4d05-41e6-9722-fdfe3acb2405/handwashing_8_wide_eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg",
            url: null,
            attribution: "Someone ",
            caption: null,
            "alt-text": "alt text",
            metadata: {
              "original-url":
                "https://media.npr.org/assets/img/2020/03/20/handwashing-8_wide-eb1f71948877c7a15d27a82cd7d90b324b956a89.jpg?s=1400",
              width: 1400,
              height: 787
            }
          }
        },
        attributes: {}
      }
    }
  ],
  url: "https://ace-web.qtstage.io/news/health/doctor-speaks-out-to-save-elderly-patients-from-the-coronavirus",
  "story-version-id": "de3cafbc-1387-4df3-8c87-639b3ac5db53",
  "content-type": "story",
  "content-updated-at": 1585646855804,
  "author-id": 190071,
  "owner-id": 190071,
  "linked-story-ids": ["78bae455-84d9-4217-9208-6927ed5989e1"],
  access: "subscription",
  "promotional-message": '<p>text <a href="https://ace-web.qtstage.io/">promotional</a> message</p>',
  "asana-project-id": null,
  "first-published-at": 1584680675181,
  "hero-image-caption": "This is the image element caption.",
  version: 29,
  "story-template": "text",
  "sequence-no": null,
  "created-at": 1585646849224,
  authors: [
    {
      id: 190071,
      name: "tester",
      slug: "tester",
      "avatar-url": null,
      "avatar-s3-key": null,
      "twitter-handle": null,
      bio: null,
      "contributor-role": {
        id: 873,
        name: "Author"
      }
    }
  ],
  metadata: {
    "card-share": {
      shareable: false
    }
  },
  "publish-at": null,
  "assignee-name": "tester"
};

export { allElementsStory };
