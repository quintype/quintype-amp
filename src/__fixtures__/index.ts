import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
