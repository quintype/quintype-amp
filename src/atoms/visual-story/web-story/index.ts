import { WebStory } from "./web-story";
export { WebStory };
