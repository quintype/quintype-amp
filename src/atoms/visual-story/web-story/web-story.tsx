import React, { Fragment } from "react";
import get from "lodash/get";
import Helmet from "react-helmet";
import { withStoryAndConfig } from "../../../context";
import { WebStoryTypes } from "./types";

const WebStoryBase = ({ story, config, children }: WebStoryTypes) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-story" src="https://cdn.ampproject.org/v0/amp-story-1.0.js" />
      </Helmet>
      <amp-story
        standalone={undefined}
        title={story.headline}
        publisher={get(config, ["publisherConfig", "publisher-settings", "title"], "")}
        publisher-logo-src={get(config, ["publisherConfig", "publisher-settings", "publisher-logo", "url"], "")}
        poster-portrait-src={story["hero-image-s3-key"]}>
        {children}
      </amp-story>
    </Fragment>
  );
};

export const WebStory = withStoryAndConfig(WebStoryBase);
