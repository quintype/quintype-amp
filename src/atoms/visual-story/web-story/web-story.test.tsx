import React from "react";
import { visualStory, config } from "../../../__fixtures__";
import { WebStoryBase } from "./web-story";
import { shallow } from "enzyme";

describe("webstory component", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <WebStoryBase story={visualStory} config={config}>
        <div>child</div>
      </WebStoryBase>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
