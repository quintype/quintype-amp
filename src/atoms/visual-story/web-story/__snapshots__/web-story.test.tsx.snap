// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webstory component should match snapshot 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-story"
      src="https://cdn.ampproject.org/v0/amp-story-1.0.js"
    />
  </HelmetWrapper>
  <amp-story
    data-test-id="visual-story"
    poster-portrait-src="//gumlet.assettype.com/amplib/2020-10/6adaf26f-3b7b-45e0-9973-6f99abeecf5d/erin_agius_I7TLmA4j0c8_unsplash.jpg?w=640&mode=crop&ar=3:4"
    publisher="Newslaundry"
    publisher-logo-src="https://images.assettype.com/thecue/2020-01/fdf230c2-d529-4da6-86ad-535e54adfad8/TheCue_Logo_dark.png"
    standalone=""
    title="Pets are awesome"
  >
    <div>
      child
    </div>
  </amp-story>
</Fragment>
`;
